internal/impl/tmpl/envrc.tmpl

```diff
@@ -4,13 +4,13 @@
 use_devbox() {
     watch_file devbox.json
     if [ -f .devbox/gen/shell.nix ]; then
+        DEVBOX_SHELL_ENABLED_BACKUP=$DEVBOX_SHELL_ENABLED
```

Comment:
I don't fully understand how this solves the shell problem? maybe because I'm not familiar with the direnv config language / model

Reply:
@mohsenari I'm also not following. I thought the reason `devbox shell` doesn't work is because `DEVBOX_SHELL_ENABLED` is true and we don't allow shell-in-shell so it fails, like below:
```
❯ devbox shell
Installing nix packages. This may take a while... done.
Starting a devbox shell...

(devbox) ❯ devbox shell
Error: You are already in an active devbox shell.
Run 'exit' before calling devbox shell again. Shell inception is not supported.
```

So, I thought your fix would `unset DEVBOX_SHELL_ENABLED`  after `$(devbox shell --print-env)` prior to `eval` it. But then it seems `devbox shell --print-env` just prints `PATH`?
```
testdata/go/go-1.19 ❯ devbox shell --print-env
export PATH="/Users/savil/code/jetpack/devbox/examples/testdata/go/go-1.19/.devbox/nix/profile/default/bin:$PATH"%
```

so it seem that I don't understand either what was going wrong in the first place....

Reply:
@savil  I changed the output of `devbox shell --print-env` to print env variables that are needed to be set for plugins. If you compile from main you'll see (https://github.com/jetpack-io/devbox/blob/main/internal/boxcli/shell.go#L53).

@ipince I added `use nix .devbox/gen/shell.nix` in `.envrc`. This line sets up the computed environment for devbox shell which includes `DEVBOX_SHELL_ENABLED`. That means if I cd into the directory direnv will set `DEVBOX_SHELL_ENABLED=1` without the user actually being in devbox shell. 
So I have to save the original value before the `use nix` line. Then afterwards I put the original value back. This way both in shell and outside of shell direnv works as expected.